store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreGraphIDsTest.java

```diff
@@ -0,0 +1,248 @@
+                                .then(new Limit<>(10))
+                                .build())
+                        .graphIdsCSV(GRAPH_ID_A)
+                        .build(), contextBlankUser());
+
+        // Then
+        assertThat(results).hasSize(1);
+        assertThat(results.iterator().next().getProperties()).containsEntry(PROPERTY_1, 1);
+    }
+
+    @Test
+    public void shouldSelectGraphUsingGraphIdsOptionWithinLongOperationChain() throws Exception {
+        // Given
+        final Entity secondEntity = new Entity.Builder()
+                .group(GROUP_BASIC_ENTITY)
+                .vertex(SOURCE_BASIC)
+                .property(PROPERTY_1, 2)
+                .build();
+        final Entity thirdEntity = new Entity.Builder()
+                .group(GROUP_BASIC_ENTITY)
+                .vertex(SOURCE_BASIC)
+                .property(PROPERTY_1, 3)
+                .build();
+
+        federatedStore.execute(
+                new AddElements.Builder()
+                        .input(secondEntity)
+                        .option(DEPRECATED_GRAPHIDS_OPTION, GRAPH_ID_A)
+                        .build(), contextBlankUser());
+        federatedStore.execute(
+                new AddElements.Builder()
+                        .input(thirdEntity)
+                        .option(DEPRECATED_GRAPHIDS_OPTION, GRAPH_ID_B)
+                        .build(), contextBlankUser());
+
+        // When
+        Iterable<? extends Element> results = (Iterable<? extends Element>) federatedStore.execute(
+                new OperationChain.Builder()
+                        .first(new GetElements.Builder()
+                                .input(BASIC_VERTEX)
+                                .option(DEPRECATED_GRAPHIDS_OPTION, GRAPH_ID_A)
+                                .build())
+                        .then(new ExportToSet.Builder<>()
+                                .key("results")
+                                .build())
+                        .then(new DiscardOutput())
+                        .then(new GetElements.Builder()
+                                .input(SOURCE_BASIC)
+                                .option(DEPRECATED_GRAPHIDS_OPTION, GRAPH_ID_B)
+                                .build())
+                        .then(new ExportToSet.Builder<>()
+                                .key("results")
+                                .build())
+                        .then(new DiscardOutput())
+                        .then(new GetSetExport.Builder()
+                                .key("results")
+                                .build())
+                        .build(), contextBlankUser());
+
+        // Then
+        assertThat(results).hasSize(2);
+        assertThat((Iterable<Entity>) results).containsExactlyInAnyOrder(EXPECTED_ENTITY, thirdEntity);
+    }
+
+    @Test
+    public void shouldNotCorrectlySelectGraphUsingGraphIdsOptionOnOperationChain() throws Exception {
```

Comment:
I'm confused by the name of this test.

> should Not Correctly Select Graph Using GraphIds Option On OperationChain

Does this mean the same as "should Not Select Graph Using GraphIds Option On OperationChain" or "should Ignore Select Graph Using GraphIds Option On OperationChain"?

Reply:
Both! Wording is subjective and awkward here, I am trying to convey that the graphIds options is ignored when set on an OperationChain and that this is weird/incorrect/breaking behaviour but we are accepting it for now

Reply:
I see. Could this be changed to "should ignore historic graphIds option on operationchain", or something similar which implies why this is not doing anything. Alternatively, could add a comment under the line explaining this.